libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/ApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Bot.Builder.Adapters;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using System;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Integration.AspNet.Core
+{
+    public static class ApplicationBuilderExtensions
+    {
+        private static readonly JsonSerializer ActivitySerializer = JsonSerializer.Create();
+
+        public static IApplicationBuilder UseBotFramework(this IApplicationBuilder applicationBuilder)
+        {
+            var options = applicationBuilder.ApplicationServices.GetRequiredService<IOptions<BotFrameworkOptions>>().Value;
+
+            var botFrameworkAdapter = new BotFrameworkAdapter(options.ApplicationId, options.ApplicationPassword);
+
+            foreach (var middleware in options.Middleware)
+            {
+                botFrameworkAdapter.Use(middleware);
+            }
```

Comment:
Why wouldn't this whole thing be the depedency injected object, instead of the BotFrameworkOptions?